src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -154,6 +158,246 @@ def coe_add_monoid_hom : finite_measure α →+ measure α :=
 instance {α : Type*} [measurable_space α] : module ℝ≥0 (finite_measure α) :=
 function.injective.module _ coe_add_monoid_hom finite_measure.coe_injective coe_smul
 
+variables [topological_space α]
+
+/-- The pairing of a finite (Borel) measure `μ` with a nonnegative bounded continuous
+function is obtained by (Lebesgue) integrating the (test) function against the measure. This
+is `finite_measure.test_against'`. -/
+abbreviation test_against_nn (μ : finite_measure α) (f : α →ᵇ nnreal) : ℝ≥0 :=
+(lintegral (μ : measure α) ((coe : ℝ≥0 → ℝ≥0∞) ∘ f)).to_nnreal
+
+-- I believe the formulation is generally useful, except maybe the exact form
+-- of the assumption `f_bdd`.
+-- Where to place?
+lemma _root_.is_finite_measure.lintegral_lt_top_of_bounded_to_ennreal {α : Type*}
+  [measurable_space α] (μ : measure α) [μ_fin : is_finite_measure μ]
+  {f : α → ℝ≥0∞} (f_bdd : ∃ (c : ℝ≥0), ∀ (x : α), f(x) ≤ c) :
+  lintegral μ f < ⊤ :=
+begin
+  cases f_bdd with c hc,
+  have le : f ≤ (λ (x : α), (c : ℝ≥0∞)) := hc,
+  apply lt_of_le_of_lt (@lintegral_mono _ _ μ _ _ le),
+  rw lintegral_const,
+  exact ennreal.mul_lt_top ennreal.coe_lt_top.ne μ_fin.measure_univ_lt_top.ne,
+end
+
+-- I believe these two could be useful... (1) (Only the second one is used below, though.)
+-- Where to place?
+lemma _root_.nnreal.val_eq_dist_zero : ∀ (z : ℝ≥0), (z : ℝ) = dist 0 z :=
+by { intros z, simp only [nnreal.dist_eq, nnreal.coe_zero, zero_sub, nnreal.abs_eq, abs_neg], }
+
+-- I believe these two could be useful... (2)
+-- Where to place?
+lemma _root_.nnreal.val_eq_dist_zero' : ∀ (z : ℝ≥0), (z : ℝ) = dist z 0 :=
+by { intros z, have key := nnreal.val_eq_dist_zero z, rwa dist_comm at key, }
+
+-- Is there a simpler way for the following? I think this might be useful (it is used below).
+-- Where to place?
+lemma _root_.nnreal.le_add_dist (a b : ℝ≥0) : a ≤ b + (dist a b).to_nnreal :=
```

Comment:
I see a lot of `(dist a b).to_nnreal`. Would it make sense to use `nndist` instead?

Reply:
Thank you! I will try this.